src/periphery/MgvReader.sol

```diff
@@ -365,75 +393,69 @@ contract MgvReader {
   function openMarkets(uint from, uint maxLen, bool withConfig)
     public
     view
-    returns (address[2][] memory markets, MarketConfig[] memory configs)
+    returns (Market[] memory markets, MarketConfig[] memory configs)
   {
     uint numMarkets = _openMarkets.length;
     if (from + maxLen > numMarkets) {
       maxLen = numMarkets - from;
     }
-    markets = new address[2][](maxLen);
+    markets = new Market[](maxLen);
     configs = new MarketConfig[](withConfig ? maxLen : 0);
     unchecked {
       for (uint i = 0; i < maxLen; ++i) {
-        address tkn0 = _openMarkets[from + i][0];
-        address tkn1 = _openMarkets[from + i][1];
-        markets[i] = [tkn0, tkn1];
+        address tkn0 = _openMarkets[from + i].tkn0;
+        address tkn1 = _openMarkets[from + i].tkn1;
+        uint tickScale = _openMarkets[from + i].tickScale;
+        // copy
+        markets[i] = Market({tkn0: tkn0, tkn1: tkn1, tickScale: tickScale});
 
         if (withConfig) {
-          configs[i].config01 = localUnpacked(tkn0, tkn1);
-          configs[i].config10 = localUnpacked(tkn1, tkn0);
+          configs[i].config01 = localUnpacked(toOLKey(markets[i]));
+          configs[i].config10 = localUnpacked(toOLKey(flipped(markets[i])));
         }
       }
     }
   }
 
-  /// @notice We choose a canonical orientation for all markets based on the numerical values of their token addresses. That way we can uniquely identify a market with two addresses given in any order.
-  /// @return address the lowest of the given arguments (numerically)
-  /// @return address the highest of the given arguments (numerically)
-  function order(address tkn0, address tkn1) public pure returns (address, address) {
-    return uint160(tkn0) < uint160(tkn1) ? (tkn0, tkn1) : (tkn1, tkn0);
-  }
-
-  /// @param tkn0 one token of the market
-  /// @param tkn1 another token of the market
+  /// @param market the market
   /// @return bool Whether the {tkn0,tkn1} market is open.
   /// @dev May not reflect the true state of the market on Mangrove if `updateMarket` was not called recently enough.
-  function isMarketOpen(address tkn0, address tkn1) external view returns (bool) {
-    (tkn0, tkn1) = order(tkn0, tkn1);
-    return marketPositions[tkn0][tkn1] > 0;
+  function isMarketOpen(Market memory market) external view returns (bool) {
```

Comment:
I seem to remember we allowed half-open markets. This seems to not allow it. What am I missing?

Reply:
`tkn0,tkn1` are canonically ordered so I think this does allow half-open markets?

Reply:
Sorry I had forgotten what marketPositions contained :) you are right.